File: 2nd_lab/2nd_task/2nd task.swift
import Foundation

func main() {
    var Bottles: Int
    var TradeNeeded: Int
    print("Введите количество банок с водой: ")
    Bottles = Int(readLine()!)!
    print("Введите количество пустых банок для обмена на 1 полную:")
    TradeNeeded = Int(readLine()!)!

    var waterConsumed = 0 // Количество выпитой воды
    var iterations = 0 // Количество итераций
    var emptyBottles = Bottles // Количество пустых банок воды

    // Процесс обмена пустых банок на полные
    if Bottles <= 0 || TradeNeeded <= 0 {
        print("Ошибка! Неверно введенные данные!")
    } else {
        while emptyBottles >= TradeNeeded{
            let exchangedBottles = emptyBottles / TradeNeeded // Количество полных банок, которые можно получить
            waterConsumed += exchangedBottles // Добавляем выпитую воду
            emptyBottles = emptyBottles % TradeNeeded + exchangedBottles // Обновляем количество пустых банок
            iterations += 1 // Увеличиваем счетчик итераций
            print("Итерация №\(iterations): Выпито \(waterConsumed) банок воды. Пустых банок осталось: \(emptyBottles)")
        }
    }
    // Вывод результатов
    print("Максимальное количество банок, которое можно выпить: \(waterConsumed)")
    print("Общее количество итераций: \(iterations)")
}

main()

File: 2nd_lab/2nd_task/2nd task.js
let readline = require('readline');

let rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Введите количество банок с водой: ", (bottles) => {
    rl.question("Введите количество пустых банок для обмена на 1 полную: ", (tradeNeeded) => {
        bottles = parseInt(bottles);
        tradeNeeded = parseInt(tradeNeeded);

        let waterConsumed = bottles; // Количество выпитой воды
        let iterations = 0; // Количество итераций
        let emptyBottles = bottles; // Количество пустых банок воды

        // Процесс обмена пустых банок на полные
        if (bottles <= 0 || tradeNeeded <= 0) {
            console.log("Ошибка! Неверно введенные данные!");
        } else {
            while (emptyBottles >= tradeNeeded) {
                let exchangedBottles = Math.floor(emptyBottles / tradeNeeded); // Количество полных банок, которые можно получить
                waterConsumed += exchangedBottles; // Добавляем выпитую воду
                emptyBottles = emptyBottles % tradeNeeded + exchangedBottles; // Обновляем количество пустых банок
                iterations++; // Увеличиваем счетчик итераций
                console.log('Итерация №${iterations}: Выпито ${waterConsumed} банок воды. Пустых банок осталось: ${emptyBottles}');
            }
        }

        // Вывод результатов
        console.log('Максимальное количество банок, которое можно выпить: ${waterConsumed}');
        console.log('Общее количество итераций: ${iterations}');

        rl.close();
    });
});
